Framework/Algorithms/src/CreateBootstrapWorkspaces.cpp

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright &copy; 2025 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include <cmath>
+#include <random>
+#include <string>
+
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/Progress.h"
+#include "MantidAPI/Workspace.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAlgorithms/CreateBootstrapWorkspaces.h"
+
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidAPI/WorkspaceGroup_fwd.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/ListValidator.h"
+#include "MantidKernel/Logger.h"
+#include "MantidKernel/normal_distribution.h"
+#include "MantidKernel/uniform_int_distribution.h"
+
+using Mantid::Kernel::BoundedValidator;
+using Mantid::Kernel::Direction;
+using Mantid::Kernel::StringListValidator;
+using namespace Mantid::API;
+
+namespace {
+Mantid::Kernel::Logger g_log("CreateBootstrapWorkspaces");
+}
+
+namespace Mantid::Algorithms {
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(CreateBootstrapWorkspaces)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string CreateBootstrapWorkspaces::name() const { return "CreateBootstrapWorkspaces"; }
+
+/// Algorithm's version for identification. @see Algorithm::version
+int CreateBootstrapWorkspaces::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string CreateBootstrapWorkspaces::category() const { return "Simulation"; }
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string CreateBootstrapWorkspaces::summary() const {
+  return "Creates a randomly simulated workspace by sampling from the probability distribution of input data.";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void CreateBootstrapWorkspaces::init() {
+
+  declareProperty(std::make_unique<WorkspaceProperty<MatrixWorkspace>>("InputWorkspace", "", Direction::Input),
+                  "Input Workspace containing data to be simulated");
+
+  auto mustBePositiveOrEqualToZero = std::make_shared<BoundedValidator<int>>();
+  mustBePositiveOrEqualToZero->setLower(0);
+  declareProperty("Seed", 32, mustBePositiveOrEqualToZero,
+                  "Integer seed that initialises the random-number generator, for reproducibility");
+
+  auto mustBePositive = std::make_shared<BoundedValidator<int>>();
+  mustBePositive->setLower(1);
+  declareProperty("NumberOfReplicas", 100, mustBePositive,
+                  "Number of Monte Carlo events to simulate. Defaults to integral of input workspace if 0.");
```

Comment:
Should this be updated to refer to the "ErrorSampling", "SpectraSampling" modes, i.e. Gaussian or sampling?